File: app/src/main/java/org/alie/paintgradientwork/view/gradient/SweepGradientView.java
package org.alie.paintgradientwork.view.gradient;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.SweepGradient;
import android.util.AttributeSet;
import android.view.View;

import androidx.annotation.Nullable;

/**
 * Created by Alie on 2019/3/18.
 * 类描述
 * 版本
 */
public class SweepGradientView extends View {
    private Paint mPaint;
    private int[] mColors = {Color.RED,Color.YELLOW,Color.BLUE,Color.GREEN};

    public SweepGradientView(Context context) {
        super(context);
        mPaint = new Paint();
        mPaint.setAntiAlias(true);
    }

    public SweepGradientView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        mPaint = new Paint();
        mPaint.setAntiAlias(true);
    }

    public SweepGradientView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        mPaint = new Paint();
        mPaint.setAntiAlias(true);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        SweepGradient sweepGradient = new SweepGradient(getWidth()/2,getHeight()/2,mColors,null);
        mPaint.setShader(sweepGradient);
        canvas.drawCircle(getWidth()/2,getHeight()/2,getWidth()/2,mPaint);
    }
}

File: app/src/main/java/org/alie/paintgradientwork/kt/gradient/SweepGradientView.kt
package org.alie.paintgradientwork.kt.gradient

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.graphics.SweepGradient
import android.util.AttributeSet
import android.view.View

/**
 * Created by Administrator on 2020/8/9 0009.
 * 类描述
 * 版本
 */
class SweepGradientView(context: Context, attributeSet: AttributeSet?, defStyle: Int) :
    View(context, attributeSet, defStyle) {

    var mPaint: Paint? = null
    var sweepGradient: SweepGradient? = null
    var mColors = intArrayOf(Color.GREEN, Color.TRANSPARENT)

    init {
        sweepGradient = SweepGradient(150F, 150F, mColors, null)
        mPaint = Paint()
        mPaint?.let {
            it.isAntiAlias = true
            it.shader = sweepGradient
        }
    }

    constructor(context: Context, attributeSet: AttributeSet) : this(context, attributeSet, 0)
    constructor(context: Context) : this(context, null, 0)

    override fun onDraw(canvas: Canvas?) {
        super.onDraw(canvas)
        canvas?.drawCircle(150F, 150F, 150F / 2, mPaint!!)
    }
}